components/authentication/prune-has.yaml

```diff
@@ -0,0 +1,43 @@
+kind: ClusterRole
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  name: has-pruner
+rules:
+  - verbs:
+      - list
+      - get
+      - delete
+      - update
+    apiGroups:
+      - 'appstudio.redhat.com'
+    resources:
+      - applications
+      - components
+      - componentdetectionqueries
+  - verbs:
+      - create # allows addition of credentials only.
```

Comment:
throw in `list` ?

Reply:
oh this is for tekton

Reply:
nope, secrets. 

You don't need secrets and tekton in the ClusterRole, do you?

Reply:
Done.